modules/demo/api-server/routes/graphology/index.js

```diff
@@ -0,0 +1,393 @@
+          const message  = 'File is available';
+          result.message = message;
+          result.success = true;
+          try {
+            const graphology = await fastify.readLargeGraphDemo(path);
+            await fastify.setDataframe('nodes', graphology['nodes']);
+            await fastify.setDataframe('edges', graphology['edges']);
+            await fastify.setDataframe('options', graphology['options']);
+            result.message = 'File read onto GPU.';
+          } catch (e) {
+            result.success    = false;
+            result.message    = e;
+            result.statusCode = 500;
+          }
+        }
+      } catch (e) {
+        result.success    = false;
+        result.message    = e.message;
+        result.statusCode = 404;
+      };
+      await reply.code(result.statusCode).send(result);
+    }
+  });
+
+  fastify.route({
+    method: 'POST',
+    url: '/read_json',
+    schema: {
+      querystring: {filename: {type: 'string'}, 'rootkeys': {type: 'array'}},
+      response: {
+        200: {
+          type: 'object',
+          properties:
+            {success: {type: 'boolean'}, message: {type: 'string'}, params: {type: 'string'}}
+        }
+      }
+    },
+    handler: async (request, reply) => {
+      const query = request.query;
+      let result  = {
+        'params': JSON.stringify(query),
+        success: false,
+        message: 'Finding file.',
+        statusCode: 200
+      };
+      try {
+        let path = undefined;
+        if (query.filename === undefined) {
+          result.message    = 'Parameter filename is required';
+          result.statusCode = 400;
+        } else if (query.filename.search('http') != -1) {
+          result.message    = 'Remote files not supported yet.';
+          result.statusCode = 406;
+        } else {
+          result.message = 'File is not remote';
+          // does the file exist?
+          const path = Path.join(__dirname, query.filename);
+          console.log('Does the file exist?');
+          const stats    = await Stat(path);
+          const message  = 'File is available';
+          result.message = message;
+          result.success = true;
+          try {
+            const graphology = await fastify.readGraphology(path);
+            await fastify.setDataframe('nodes', graphology['nodes']);
+            await fastify.setDataframe('edges', graphology['edges']);
+            await fastify.setDataframe('clusters', graphology['clusters']);
+            await fastify.setDataframe('tags', graphology['tags']);
+            result.message = 'File read onto GPU.';
+          } catch (e) {
+            result.success    = false;
+            result.message    = e;
+            result.statusCode = 500;
+          }
+        }
+      } catch (e) {
+        result.success    = false;
+        result.message    = e.message;
+        result.statusCode = 404;
+      };
+      await reply.code(result.statusCode).send(result);
```

Comment:
Any advice on collapsing the above `if` farm into something more functional?

Reply:
I feel like this conditional is accurate and concise. The only functional idiom I could think would replace it is something like an Option monad, but that's not easy without a library. A+ code imo.